components/network/isc-dhcp/patches/005-fno-common.patch

```diff
@@ -0,0 +1,76 @@
+--- dhcp-4.4.2/server/mdb.c.orig	2020-05-17 09:24:45.683862500 +0000
++++ dhcp-4.4.2/server/mdb.c	2020-05-17 09:24:54.377630988 +0000
+@@ -67,8 +67,6 @@
+ 
+ int numclasseswritten;
+ 
+-omapi_object_type_t *dhcp_type_host;
+-
+ isc_result_t enter_class(cd, dynamicp, commit)
+ 	struct class *cd;
+ 	int dynamicp;
+--- dhcp-4.4.2/client/dhclient.c.orig	2020-05-17 09:20:00.076824364 +0000
++++ dhcp-4.4.2/client/dhclient.c	2020-05-17 09:23:57.744717752 +0000
+@@ -83,8 +83,6 @@
+ static const char url [] = "For info, please visit https://www.isc.org/software/dhcp/";
+ #endif /* UNIT_TEST */
+ 
+-u_int16_t local_port = 0;
+-u_int16_t remote_port = 0;
+ #if defined(DHCPv6) && defined(DHCP4o6)
+ int dhcp4o6_state = -1; /* -1 = stopped, 0 = polling, 1 = started */
+ #endif
+--- dhcp-4.4.2/keama/keama.h.orig	2020-05-17 09:26:40.704314610 +0000
++++ dhcp-4.4.2/keama/keama.h	2020-05-17 09:36:19.201790252 +0000
+@@ -31,11 +31,13 @@
+ #include <time.h>
+ 
+ /* Resolution of FQDNs into IPv4 addresses */
+-enum resolve { 
++typedef enum resolve { 
+ 	perform = 0,	/* resolve */
+ 	fatal,		/* raise a fatal error */
+ 	pass		/* pass the string wth a warning */
+-} resolve;
++} resolve_t;
++
++extern resolve_t resolve;
+ 
+ /* From includes/dhcp.h */
+ 
+@@ -58,7 +60,9 @@
+ 
+ /* A parsing context. */
+ 
+-TAILQ_HEAD(parses, parse) parses;
++typedef TAILQ_HEAD(parses, parse) parses_t;
++
++extern parses_t parses;
+ 
+ struct parse {
+ 	int lexline;
+--- dhcp-4.4.2/keama/keama.c.orig	2020-05-17 09:32:35.540360334 +0000
++++ dhcp-4.4.2/keama/keama.c	2020-05-17 09:36:33.254324494 +0000
+@@ -37,6 +37,9 @@
+ 			" [-l hook-library-path]" \
+ 			" [-i input-file] [-o output-file]\n"
+ 
++resolve_t resolve;
++parses_t parses;
++
+ static void
+ usage(const char *sfmt, const char *sarg) {
+ 	if (sfmt != NULL) {
+--- dhcp-4.4.2/relay/dhcrelay.c.orig	2020-05-17 09:18:51.856433510 +0000
++++ dhcp-4.4.2/relay/dhcrelay.c	2020-05-17 09:19:12.926920464 +0000
+@@ -95,8 +95,8 @@
+        forward_untouched,	/* Forward without changes. */
+        discard } agent_relay_mode = forward_and_replace;
+ 
+-u_int16_t local_port;
+-u_int16_t remote_port;
++extern u_int16_t local_port;
++extern u_int16_t remote_port;
+ 
```

Comment:
There are still two declarations of local_port and remote_port:
./common/discover.c:u_int16_t local_port;
./omapip/connection.c:		   u_int16_t local_port

Reply:
Yes they are the symbols generated in the library. I'll recheck the build.